### PR TITLE
Zones Selector sorted by zone names

### PR DESCRIPTION
**Pull request by AleziaKurdis**
This change makes the Zones Selector sorted by zone names.
To address issue: renderWithZones: zone list incorrectly sorted. #354
(But this doesn't fix the problem with the scrolling)

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-5b2f1da_3a0c4e14.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-5b2f1da_3a0c4e14.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-5b2f1da.tar_3a0c4e14.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-5b2f1da.tar_3a0c4e14.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-5b2f1da_3a0c4e14.exe

**Comment by kasenvr** [comment]
Zones appear to be sorting correctly. :)

File: scripts/system/create/edit.js [hunk 1]
```diff
@@ -2905,7 +2905,19 @@ function getExistingZoneList() {
         };
         listExistingZones.push(thisZone);
     }
+    listExistingZones.sort(GetSortOrder("name"));
```
Comment (ctrlaltdavid) [inline scripts/system/create/edit.js:2908]:
```suggestion
    listExistingZones.sort(zoneSortOrder);
```
Two things here:
- Sort function name should start with a lower case letter.
- Not passing the sort property into the function may speed up the sort?

(Set `var zoneSortProperty = "name";` outside function and update function code to use this variable.)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-851c12e_3023924c.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-851c12e_3023924c.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-851c12e.tar_3023924c.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-851c12e.tar_3023924c.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-851c12e_3023924c.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-d219f64.tar_e6d092ae.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-d219f64.tar_e6d092ae.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-d219f64_e6d092ae.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-d219f64_e6d092ae.exe

**Comment by SilverfishVR** [comment]
Tested with _ProjectAthena-Alpha-PR369-d219f64_e6d092ae.exe_

Zones are sorted capital letters first, then alphabetically, is this intentional?
Screenshot 2020-05-23 19 51 31
 Render with zones seems to work, even render zones with zones :-)

**Comment by kasenvr** [comment]
We probably want it to be a case insensitive sort, so perhaps converting the strings using the `toUppercase()` function when comparing them might work.

**Comment by SilverfishVR** [comment]
Tested: _ProjectAthena-Alpha-PR369-f21297f_b7393b23.exe_
Screenshot 2020-05-26 19 01 20

File: scripts/system/create/edit.js [hunk 2]
```diff
@@ -2905,7 +2905,21 @@ function getExistingZoneList() {
         };
         listExistingZones.push(thisZone);
     }
+    listExistingZones.sort(zoneSortOrder());
```
Comment (ctrlaltdavid) [inline scripts/system/create/edit.js:2908]:
```suggestion
    listExistingZones.sort(zoneSortOrder);
```
Just need to pass in the function identify, not call it.

**Comment by thoys** [comment]
> Tested: _ProjectAthena-Alpha-PR369-f21297f_b7393b23.exe_
> Screenshot 2020-05-26 19 01 20

Seems a bit inconsistent there,     A -> a -> b -> B    ,  wouldn't you expect it to be A -> a -> B -> b  or a -> A -> b -> B  ?

**Comment by thoys** [comment]
Ah I see it is because you ignore the case completely in the comparison. I remember doing a sort in the entity list, perhaps there is a hint in there for a better way to tackle this.

**Comment by AleziaKurdis** [comment]
Is this would do the job for you?

[{"name":"a1","key":"a1"},
{"name":"A1","key":"A1"},
{"name":"à1","key":"à1"},
{"name":"À1","key":"À1"},
{"name":"b1","key":"b1"},
{"name":"B1","key":"B1"},
{"name":"c1","key":"c1"},
{"name":"C1","key":"C1"},
{"name":"e1","key":"e1"},
{"name":"E1","key":"E1"}]

Let me know.

**Comment by ctrlaltdavid** [comment]
Perhaps try String.localeCompare(), in the zoneSortOrder() function.

**Comment by AleziaKurdis** [comment]
"localeCompare" works fine when I try in a web browser. But in the Create App, it doesn't return the same order. Actually it does a worse result.

So I tried this:
function zoneSortOrder(a, b) {
    var nameA = a.name.toUpperCase();
    var nameB = b.name.toUpperCase();
    if (nameA > nameB) {
        return 1;    
    } else if (nameA < nameB) {
        return -1;
    }
    if (a.name > b.name) {
        return 1;    
    } else if (a.name < b.name) {
        return -1;
    }
    return 0;
}

and it's already better, now the ordering is predictable:
Sort

The accents char are however not considered as we would expect.
I have not found any simple way to remove the accent that would work in **every** language.

I also tried with other charsets like Japanese, the ordering is working.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-c44ea9b_0b330dfc.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-c44ea9b.tar_0b330dfc.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-c44ea9b_0b330dfc.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-c44ea9b.tar_0b330dfc.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR369/ProjectAthena-Alpha-PR369-c44ea9b_0b330dfc.exe

**Pull request merged**